target/linux/ath79/dts/qcn5502_netgear_ex7300v2.dts

```diff
@@ -0,0 +1,214 @@
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		sck-gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		mosi-gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+		num-chipselects = <0>;
+
+		led_gpio: led_gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <1>;
+			spi-max-frequency = <500000>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power_green: power_green {
+			label = "green:power";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_amber: power_amber {
+			label = "amber:power";
+			gpios = <&led_gpio 7 GPIO_ACTIVE_LOW>;
+		};
+
+		wps_green {
+			label = "green:wps";
+			gpios = <&led_gpio 6 GPIO_ACTIVE_LOW>;
+		};
+
+		router_red {
+			label = "red:router";
+			gpios = <&led_gpio 5 GPIO_ACTIVE_LOW>;
+		};
+
+		router_green {
+			label = "green:router";
+			gpios = <&led_gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		client_red {
+			label = "red:client";
+			gpios = <&led_gpio 3 GPIO_ACTIVE_LOW>;
+		};
+
+		client_green {
+			label = "green:client";
+			gpios = <&led_gpio 2 GPIO_ACTIVE_LOW>;
+		};
+
+		left_blue {
+			label = "blue:left";
+			gpios = <&led_gpio 1 GPIO_ACTIVE_LOW>;
+		};
+
+		right_blue {
+			label = "blue:right";
+			gpios = <&led_gpio 0 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "WPS button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		extender_apmode {
+			label = "EXTENDER/APMODE switch";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,code = <BTN_0>;
+			linux,input-type = <EV_SW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pll {
+	clocks = <&extosc>;
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+
+			partition@50000 {
+				label = "config";
+				reg = <0x050000 0x010000>;
+			};
+
+			partition@60000 {
+				label = "pot";
+				reg = <0x060000 0x010000>;
+			};
+
+			partition@70000 {
+				label = "firmware";
+				reg = <0x70000 0xe30000>;
+				compatible = "denx,uimage";
+			};
+
+			partition@ea0000 {
+				label = "rae";
+				reg = <0xea0000 0x100000>;
+			};
+
+			partition@fa0000 {
+				label = "oopsdump";
+				reg = <0xfa0000 0x40000>;
+			};
+
+			artmtd: partition@fe0000 {
+				label = "artmtd";
+				reg = <0xfe0000 0x10000>;
+			};
+
+			art: partition@ff0000 {
+				label = "art";
+				reg = <0xff0000 0x10000>;
+			};
```

Comment:
You should use read-only on most of the partitions.

Reply:
Done. Left out `config` which seems to be the config for the stock firmware, and `artmtd` because the stock firmware provides `artmtd` command for modifying that partition. Note that the stock firmware only sets read-only on `u-boot`.

Reply:
I don't really see why one would modify artmtd partition? Maybe one should use art for MAC addresses etc. then instead, as it's the real source of data which is not manipulated?

Reply:
One reason to modify artmtd is to change the stock firmware region ([link](https://forum.openwrt.org/t/netgear-r7800-how-to-unlock-wifi-region-selector-in-stock-firmware/45108)), is that a good reason?

Based on dumping the partitions, the MAC address for this device is only contained in artmtd, not art.

Reply:
I also dumped the entire flash using an external SPI programmer, and the only plaintext instance of the MAC address is in artmtd